### PR TITLE
Saving prediction results on cikm contest data

### PR DESCRIPTION
**Pull request by DavdGao**
- This PR is a temporary solution to save prediction results on the cikmcup dataset. 
- It will be updated soon.
- It should be merged after the PR #227

### CODE REVIEW

File: federatedscope/gfl/trainer/graphtrainer.py [hunk 1]
```diff
@@ -65,6 +79,16 @@ def _hook_on_batch_forward_flop_count(self, ctx):
         self.ctx.monitor.total_flops += self.ctx.monitor.flops_per_sample * \
             ctx.batch_size
 
+    def save_prediction(self, client_id, task_type):
+        y_inds, y_probs = self.ctx.test_y_inds, self.ctx.test_y_prob
+        if 'classification' in task_type:
+            y_probs = np.argmax(y_probs, axis=-1)
```
Comment (rayrayraykk) [inline federatedscope/gfl/trainer/graphtrainer.py:85]: `y_preds = np.argmax(y_probs, axis=-1)`

Comment (DavdGao) [reply federatedscope/gfl/trainer/graphtrainer.py]: Modified accordingly

File: federatedscope/gfl/trainer/graphtrainer.py [hunk 2]
```diff
@@ -69,6 +84,21 @@ def _hook_on_batch_forward_flop_count(self, ctx):
         self.ctx.monitor.total_flops += self.ctx.monitor.flops_per_sample * \
             ctx.batch_size
 
+    def save_prediction(self, client_id, task_type):
+        y_inds, y_probs = self.ctx.test_y_inds, self.ctx.test_y_prob
+        os.makedirs('../prediction', exist_ok=True)
```
Comment (joneswong) [inline federatedscope/gfl/trainer/graphtrainer.py:89]: why is "../" necessary?

**Review by joneswong — APPROVED** [review]
LGTM

File: federatedscope/gfl/trainer/graphtrainer.py (hunk 2 above)
Comment (DavdGao) [reply federatedscope/gfl/trainer/graphtrainer.py]: modified accordingly

**Pull request merged**